components/raftstore-v2/src/operation/command/admin/compact_log.rs

```diff
@@ -377,16 +416,14 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             );
             return;
         }
-        if let Some(i) = self.merge_context().and_then(|c| c.max_compact_log_index())
-            && res.compact_index > i
-        {
-            info!(
+        if res.compact_index >= self.compact_log_context().last_applying_index {
```

Comment:
How can that happen? `compact_index` comes from apply result, which implies CompactLog at index i0 is applied and i0 > compact_index, which implies i0 <= last_applying_index, so compact_index < i0 <= last_applying_index.

Reply:
Probably because compact_index is nudged at L408. I didn't bother understand the exact situation, I just fixed it to pass tests.

Reply:
But L408 is decreasing `compact_index` rather than increasing.

Reply:
Oh, apparently a test uses invalid compact log to test prepare merge is proposed.